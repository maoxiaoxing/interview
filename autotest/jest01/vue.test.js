import Vue from 'vue/dist/vue'

function renderVueComponent () {
  document.body.innerHTML = `
    <div id="app"></div>
  `
  new Vue({
    template: `
      <div id="app">
        <h1>{{ message }}</h1>
      </div>
    `,
    data: {
      message: 'hello vue'
    }
  }).$mount('#app')
}

test('vue test', () => {
  renderVueComponent()
  console.log(document.body.innerHTML)
  expect(document.body.innerHTML).toMatch(/hello vue/)
})
