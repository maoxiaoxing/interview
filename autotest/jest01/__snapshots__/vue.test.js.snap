// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
"
    <div id=\\"app\\"><h1>hello vue</h1></div>
  "
`;
